sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,451 @@
+            for l in lst:
+                rospy.logwarn("acting on: %s", l)
+            if command_type.upper() == CONST_POSITION_CHAR:
+                file_prefix = raw_input("enter file prefix:")
+                for j in lst:
+                    if 'TH' not in j.upper():
+                        if file_prefix == '':
+                            file_prefix = '_'
+                        self.test_joint(j, mode='testing', prefix=file_prefix)
+            elif command_type.upper() == CONST_EFFORT_CHAR:
+                file_prefix = raw_input("enter file prefix:")
+                for j in lst:
+                    if 'TH' not in j.upper():
+                        if 'LF' in j.upper() or 'RF' in j.upper():
+                            angle = str(float(angle)*(-1.0))
+                        self.test_joint(j, 'PWM', value=angle, prefix=file_prefix + '_plus')
+                        rospy.sleep(1)
+                        self.test_joint(j, 'PWM', value=str(float(angle)*(-1.0)), prefix=file_prefix + '_minus')
+                        rospy.sleep(1)
+
+    def switch_finger_to_effort(self, finger):
+        joints_to_change = []
+        for joint in self.requested_joints:
+            if finger in joint:
+                joints_to_change.append(joint)
+        temp_controller_helper = ControllerHelper([self._hand_prefix[0] + 'h'], [self._hand_prefix],
+                                                  [joint.lower() for joint in joints_to_change])
+        temp_controller_helper.change_hand_ctrl("effort")
+
+    def switch_finger_to_position(self, finger):
+        joints_to_change = []
+        for joint in self.requested_joints:
+            if finger.lower() in joint.lower():
+                joints_to_change.append(joint)
+        temp_controller_helper = ControllerHelper([self._hand_prefix[0] + 'h'], [self._hand_prefix],
+                                                  [joint.lower() for joint in joints_to_change])
+        temp_controller_helper.change_hand_ctrl("position")
+
+    def switch_to_effort(self):
+        self._controller_helper.change_hand_ctrl("effort")
+        self._mode = 'effort'
+
+    def switch_to_position(self):
+        self._controller_helper.change_hand_ctrl("position")
+        self._mode = 'position'
+
+    def publish_pwm(self, joint, pwm):
+        if 'effort' in self._mode:
+            rospy.loginfo("Applying a PWM of %s to joint %s", str(pwm), joint)
+            self._pwm_command_publishers[joint.upper()].publish(Float64(float(pwm)))
+        else:
+            rospy.logerr("Mode not set to effort, please change this before applying a PWM")
+
+    def test_joint(self, joint, mode='testing', value=0, sleep=3, prefix=''):
+        if '4' in joint and 'th' not in joint.lower():
+            self.free_j4(joint)
+            rospy.sleep(1.5)
+        file_prefix = prefix
+        if prefix == '':
+            file_prefix = raw_input("Name/describe the test (adds text to filename, optional)")
```

Comment:
why do we need file_prefix? is this the description of the test?

Reply:
yeah, it's an optional description you can add so that finding your data later is easier